fawltydeps/extract_declared_dependencies.py

```diff
@@ -288,6 +288,28 @@ def parse_pyproject_toml(path: Path) -> Iterator[DeclaredDependency]:
     else:
         logger.debug("%s does not contain [tool.poetry].", source)
 
+    if "dynamic" in parsed_contents.get("tool", {}).get("setuptools", {}).keys():
+        dynamic_section = parsed_contents["tool"]["setuptools"]["dynamic"]
+        dynamic_deps_files = dynamic_section.get("dependencies", {}).get("file", [])
+        dynamic_optional_deps_files = [
+            file_list
+            for optional_deps_item in [
+                optional_deps_section.get("file", {})
+                for optional_deps_section in dynamic_section.get(
+                    "optional-dependencies", {}
+                ).values()
+            ]
+            for file_list in optional_deps_item
+        ]
+        dynamic_files = dynamic_deps_files + dynamic_optional_deps_files
+        if isinstance(dynamic_files, list):
+            for req_file in dynamic_files:
+                req_file_path = Path(req_file)
```

Comment:
From this [example](https://github.com/ansible/ansible-lint/blob/f6961ffcc70748753110efb84ec6ca6dcd5fa128/pyproject.toml#L271-L275), it seems like the provided path is a relative one. Do we need to compute the full path of the file by joining with the basepath?

Reply:
Hmm. For most projects I expect the `pyproject.toml` to be located in the root dir of the project, so it shouldn't matter (as FD would typically run from the same dir), but, yes, I expect there are scenarios where we need to be more careful.

AFAICS, [setuptools](https://setuptools.pypa.io/en/latest/userguide/pyproject_config.html#dynamic-metadata) does not specify the base to be used for these relative paths, but I think it's fairly safe to assume that it should be the directory that contains the `pyproject.toml` file itself. Hence the last line here should probably be something like:

```python
                req_file_path = path.with_name(req_file)
```
or
```python
                req_file_path = path / req_file
```

Definitely something we should have a test case for....

Reply:
Thanks for the clarifications!
I adopted the second option here 
```
        req_file_path = path.parent / req_file
```
since `path` refers to `basepath/pyproject.toml`

Reply:
Addressed in https://github.com/tweag/FawltyDeps/pull/351/commits/4242069da6847018533f95b1a4af1ee86159a4a0 and https://github.com/tweag/FawltyDeps/pull/351/commits/4c350f5438802703a55cbe67bc02ca452aef021d